app/models/rbac.rb

```diff
@@ -543,8 +543,32 @@ def self.apply_scope(klass, scope)
     end
   end
 
+  # it skips filters which are substring in any other @filters
+  # @param filters string array with filters
+  # @return filtered array with only most specialized filters
+  # example:
+  # it converts:
+  # ["/belongsto/ExtManagementSystem|XX1/EmsFolder|Folder1/EmsFolder/YY1...",
+  # "/belongsto/ExtManagementSystem|XX1"
+  # "/belongsto/ExtManagementSystem|XX2"]
+  # to
+  # ["/belongsto/ExtManagementSystem|XX1/EmsFolder|Folder1/EmsFolder/YY1...",
+  # "/belongsto/ExtManagementSystem|XX2"]
+  def self.get_most_specialized_filters(filters)
+    include_substring_in_filters = lambda do |substr|
+      has_substr = filters.select do |x|
+        # Does x starts with substr ? (expect to same strings)
+        /^#{substr.gsub(/\|/, "\\|")}[^$]/.match(x)
```

Comment:
I need to skip strings which are same or do you think this is better ? maybe its more readable and I don't need to escape special string, what do you think ?
```
x.starts_with?(substr) && x!=substr
```